cmd/zpool/zpool_main.c

```diff
@@ -7253,15 +7253,59 @@ print_error_log(zpool_handle_t *zhp)
 	elem = NULL;
 	while ((elem = nvlist_next_nvpair(nverrlist, elem)) != NULL) {
 		nvlist_t *nv;
-		uint64_t dsobj, obj;
+		uint64_t dsobj, obj, data_block_size, indirect_block_size;
+		uint64_t *block_ids;
+		int64_t *indrt_levels;
+		unsigned int error_count;
 
 		verify(nvpair_value_nvlist(elem, &nv) == 0);
 		verify(nvlist_lookup_uint64(nv, ZPOOL_ERR_DATASET,
 		    &dsobj) == 0);
 		verify(nvlist_lookup_uint64(nv, ZPOOL_ERR_OBJECT,
 		    &obj) == 0);
-		zpool_obj_to_path(zhp, dsobj, obj, pathname, len);
-		(void) printf("%7s %s\n", "", pathname);
+		verify(nvlist_lookup_int64_array(nv, ZPOOL_ERR_LEVEL,
+		    &indrt_levels, &error_count) == 0);
+		verify(nvlist_lookup_uint64_array(nv, ZPOOL_ERR_BLOCKID,
+		    &block_ids, &error_count) == 0);
+
+		zpool_obj_to_path(zhp, dsobj, obj, pathname, len,
+		    &data_block_size, &indirect_block_size);
+
+		if (error_count > 0 && data_block_size != 0 &&
+		    indirect_block_size != 0) {
+			uint64_t blkptrs_in_ind
+			    = indirect_block_size/sizeof (blkptr_t);
+			uint64_t min_offset_blks
+			    = pow(blkptrs_in_ind, indrt_levels[0])*block_ids[0];
+			uint64_t max_offset_blks =
+			    pow(blkptrs_in_ind, indrt_levels[0])*block_ids[0];
+
+			/*
+			 * iterate through the error blockids and finds minimum
+			 * and maximax offset
+			 */
+			for (int i = 1; i < error_count; i++) {
+				uint64_t offset_blks
+				    = pow(blkptrs_in_ind,
+				    indrt_levels[i])*block_ids[i];
+				if (min_offset_blks > offset_blks) {
+					min_offset_blks = offset_blks;
```

Comment:
There's a macro we use in a few places for this, you could write this as `min_offset_blks = MIN(min_offset_blks, offset_blks)`